components/production/ProductionDraftWarningBanner.vue

```diff
@@ -0,0 +1,23 @@
+<template>
+  <div
+    v-if="['Draft', 'Pending', 'Approved'].includes(status)"
+    class="bg-sta-orange text-center w-full py-1"
+  >
+    {{ status }} Production Preview - This Page Is Not Public Yet
+  </div>
+</template>
+
+<script setup lang="ts">
+import ProductionStatusEnum from '~~/enums/ProductionStatusEnum';
+
+const props = defineProps({
+  production: {
+    required: true,
+    type: Object
+  }
+});
```

Comment:
With typescript, there is a better way to define this (currently "production" just has to be an object, which isn't very useful for making sure it is passed the right stuff (what if status isn't defined in it??)) - https://vuejs.org/guide/typescript/composition-api.html#typing-component-props

Something like this would be better and type-safe:
```ts suggestion
import {ProductionStatus} from '~/graphql/codegen/operations'

const props = defineProps<{
    production: {status: ProductionStatus}
}>();
```

This would make sure the object given as a prop is a object which contains a `status` key, where the value of the key in one of the possible `ProductionStatus` enums